query-builder/src/main/java/com/datastax/oss/driver/api/querybuilder/QueryBuilder.java

```diff
@@ -400,7 +400,18 @@ public static Literal literal(@Nullable Object value) {
    */
   @NonNull
   public static Literal literal(@Nullable Object value, @NonNull CodecRegistry codecRegistry) {
-    return literal(value, (value == null) ? null : codecRegistry.codecFor(value));
+    try {
+      return literal(value, (value == null) ? null : codecRegistry.codecFor(value));
+    } catch (CodecNotFoundException e) {
+      assert value != null;
+      throw new IllegalArgumentException(
+          String.format(
+              "Could not inline literal of type %s. "
+                  + "This happens because the driver doesn't know how to map it to a CQL type. "
+                  + "Try passing a TypeCodec or CodecRegistry to literal().",
```

Comment:
Maybe add: 

```
Try passing a TypeCodec or CodecRegistry to literal(), or register a TypeCodec that accepts "+ value.getClass().getName()) + "."
```

Reply:
This query builder method is completely detached from any session though (and the default registry is immutable), so even if you register your codec somewhere, you'll still have to pass the `CodecRegistry`.